discord_thread_summarizer_using_openai/main.bal

```diff
@@ -0,0 +1,75 @@
+import ballerina/io;
+import ballerina/log;
+import ballerina/time;
+
+const TYPE_FORUM_CHANEL = 15;
+const time:Seconds ONE_DAY_IN_SECONDS = 86400;
+configurable int forumChannelId = ?;
+
+function readThreads(ActiveThreads activeThreads) returns error? {
+
+    foreach ChannelThread thread in activeThreads.threads {
+
+        // check if the thread has any recent messages
+        if !check hasRecentMessages(thread) {
+            continue;
+        }
+
+        string prompt = check constructPrompt(thread);
+        io:println(prompt);
+
+        string|error generateChatCompletionResult = generateChatCompletion(prompt);
+
+        if generateChatCompletionResult is string {
+            io:println(generateChatCompletionResult);
+        } else {
+            log:printError("Error occured when connecting to OpenAI: ", generateChatCompletionResult, stackTrace = generateChatCompletionResult.stackTrace());
+        }
+        io:println("--------------------------------------------------\n\n");
+    }
+}
+
+function constructPrompt(ChannelThread thread) returns string|error {
+    string prompt = string `${PROMPT}Thread URL: ${getThreadURL(thread.guild_id, thread.id)}${"\n"}Title: ${thread.name}${"\n"}Question: `;
+    // reverse the array so that the messages are in chronological order
+    Message[] allMessages = (check getMessages(thread.id)).reverse();
+
+    boolean firstMessage = true;
+    foreach Message message in allMessages {
+        string formattedTimestamp = message.timestamp;
+        if message.content == "" {
```

Comment:
Do we need to handle these edge cases in a sample? IMO, it gets unnecessarily complicated/lengthy.

Reply:
If this isn't necessary I would rewrite this to assign the first message outside the loop and loop 1 ..< allMessages.length() in the for loop which will remove the firstMessage variable and checks.

Reply:
It is necessary, we are going to take this code and use it to summarize our Ballerina Discord threads, for now, we are adding this to the samples.

Reply:
Is this the only scenario that needs to be handled then?

Reply:
This is one of the scenarios. There might be more, this is one of the scenarios we encountered when developing this sample.

Reply:
In that case I think this may be overly complicating the sample, since IIUC, this is to give a basic overview rather than be a complete solution.

Reply:
Ack.

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/23d305292b97c4ea8fa597532d1e9aef7f65f9c4